### PR TITLE
fix: skip invalid models

### PR DESCRIPTION
**Pull request by leonardmq**
## What does this PR do?

Issue is described in more details here: https://github.com/Kiln-AI/Kiln/issues/454

TLDR: we update the model list over the air, which means the model list we publish to remote config must be backward compatible with older clients. However, we will break backward compatibility necessarily in certain cases, such as when we add a field to an `enum`, for example, when we add a new `ModelProviderName` because we add support for a new AI model provider.

Fix:
- skip the models that fail validation so older clients still get the new models they are able to support, but don't get the ones that break

## Checklists

- [x] Tests have been run locally and passed
- [x] New tests have been added to any work in /lib

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved handling of invalid or forward-incompatible fields in model and provider configurations, ensuring robust loading and filtering of data.
  * Updated remote configuration URLs to use the latest version for improved reliability.

* **Tests**
  * Added comprehensive tests covering error handling and edge cases in configuration deserialization.

* **Chores**
  * Adjusted workflow steps to generate and publish the updated configuration file format.

### CODE REVIEW

**Comment by leonardmq** [comment]
@coderabbitai review

File: libs/core/kiln_ai/adapters/remote_config_legacy.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            )
+
+            # config is now ready to be written to the permanent file
+            Path(path).write_text(
+                json.dumps(
+                    {"model_list": [m.model_dump(mode="json") for m in models]},
+                    indent=2,
+                    sort_keys=True,
+                )
+            )
```
[the remark this answers is not in the document]
Comment (leonardmq) [reply libs/core/kiln_ai/adapters/remote_config_legacy.py]: You are the best, fixed

File: libs/core/kiln_ai/adapters/test_remote_config_legacy.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+def test_deserialize_config_v0_18_format(tmp_path):
+    """Test that the v0.18 format serialization works and can be deserialized.
+
+    This test uses serialize_config_v0_18 which runs uv to install and test v0.18 compatibility.
+    Skip unless explicitly requested via environment variable.
+
+    Run from CLI: KILN_TEST_COMPATIBILITY=1 uv run python3 -m pytest libs/core/kiln_ai/adapters/test_remote_config_legacy.py::test_deserialize_config_v0_18_format -s -v
+    """
+    import os
+    import shutil
```
[the remark this answers is not in the document]
Comment (leonardmq) [reply libs/core/kiln_ai/adapters/test_remote_config_legacy.py]: Done

File: libs/core/kiln_ai/adapters/test_remote_config_legacy.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    """
+
+    # Skip unless explicitly requested via environment variable
+    if not os.environ.get("KILN_TEST_COMPATIBILITY"):
+        pytest.skip(
+            "Compatibility test skipped. Set KILN_TEST_COMPATIBILITY=1 to run this test."
+        )
+
+    import shutil
+    import subprocess
```
[the remark this answers is not in the document]
Comment (leonardmq) [reply libs/core/kiln_ai/adapters/test_remote_config_legacy.py]: Done
